### DISCUSSION TITLE
R4phd - Check all is OK

### DISCUSSION THREAD

**Discussion by giscus** [sorenoneill/r4phd#12] [General]
# R4phd - Check all is OK

Is everything in place before your R course?

https://sorenoneill.github.io/r4phd/installation_check_en.html

**Comment by Magnussen1901** [comment]
Hi Steen and Søren,
Thank you for this. The introductory material you've made is really very useful.
Checking the installation: I would think that I've installed it ok, but I get an error message, when I check it with your code. It's 
Error: object 'recommended_packages' not found
I did install both Tidyverse and the extra packages that you mentioned could be installed at a later point. Please let me know if this is important and if I should do something, thanks. 
I'm looking very much forward to the course.
Best,
Anne

**Reply by sorenoneill** [reply]
Hi Anna
Is it a case that the script won't run,  or that the script rund and tells you [WARNING] (instead of [OK]) about a missing package?

Regards
Soren

**Reply by Magnussen1901** [reply]
Hi Søren,
The complete reply is very short:

> for (p in recommended_packages) {
+   if (p %in% all_installed_packages) {
+     print(paste0("[OK] '", p, "' (package) is installed."))
+   } else {
+     print(paste0("[WARNING] '", p, "' (package) is NOT installed."))
+   }
+ }
Error: object 'recommended_packages' not found
> 

... This is where it becomes super clear that I'm an absolute beginner :-) I'm not sure what the code means, i.e. what it is it's supposed to tell me if it runs succesfully.
/ Anne

**Comment by steenharsted** [comment]
Dear Anne,

No worries.

The error message means that R can't find an object called `recommened_packages`. The `recommened_packages` object is created in these two lines:

```
 all_installed_packages <- .packages(all.available = TRUE)
recommended_packages <- c("tidyverse", "here", "skimr", "gt", "gtExtras", "gtsummary", "patchwork", "naniar", "reprex", "ggstatsplot", "easystats", "vroom")
```

Try to run the lines and then run:

```

for (p in recommended_packages) {
  if (p %in% all_installed_packages) {
    print(paste0("[OK] '", p, "' (package) is installed."))
  } else {
    print(paste0("[WARNING] '", p, "' (package) is NOT installed."))
  }
}
```

Hope it works.

**Reply by Magnussen1901** [reply]
Hi Steen,
Thank you - it's all OKs now :-)
See you Wednesday.
/ Anne

**Reply by sorenoneill** [reply]
Actually, you should just run the whole script - CTRL+SHIFT+s og click 'Source' top right hand corner

**Reply by Magnussen1901** [reply]
Thanks again.
Something completely different: I'm on Bornholm right now (vacation), and the plan is to fly back tomorrow in time for the course Wednesday. Due to bad weather, there is a small risk that the flight will be cancelled - they seem to be flying today (while the ferries are cancelled), so fingers crossed. Just so you know - I'll write if I cannot make it.
Best,
Anne

**Reply by sorenoneill** [reply]
Hi Anna
Yeah the Anemoi are really having a go this week. Hope you dont get blown of course and hope to see you tomorrow. Safe travels.
Soren and Steen
